In [1]:
import numpy as np
import pandas as pd

Considere os arrays numpy X, y abaixo:

In [2]:
from sklearn.datasets import load_iris

X = load_iris().data
y = load_iris().target

In [3]:
X[:10]

array([[5.1, 3.5, 1.4, 0.2],
       [4.9, 3. , 1.4, 0.2],
       [4.7, 3.2, 1.3, 0.2],
       [4.6, 3.1, 1.5, 0.2],
       [5. , 3.6, 1.4, 0.2],
       [5.4, 3.9, 1.7, 0.4],
       [4.6, 3.4, 1.4, 0.3],
       [5. , 3.4, 1.5, 0.2],
       [4.4, 2.9, 1.4, 0.2],
       [4.9, 3.1, 1.5, 0.1]])

In [4]:
y[:10]

array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])

__Exercício 2:__ Usando o df criano no exercício 1, calcule qual a média da variável 'X2', para cada valor de target. Dica: Use a função groupby no dataframe df para resolver.

__Exercício 3:__ Construa um dicionário mapeando os valores:
        
        0 - 'A'
        
        1 - 'B'
        
        2 - 'C'
        
Usando a função map, crie uma coluna nova no dataframe df chamada 'target_2' com o mapeamento feito. Dica: Use a função map como visto em aula.

__Exercício 4:__ 


4.1: Filtre o dataframe df onde X1 é maior do que 6 __ou__ target é diferente de 2 e chame-o de df1. Dica: usar os filtros e os operadores vistos em aula;

4.2: Filtre os primeiros 15 índices do dataframe df e chame-o de df2;

4.3: Transformando os dataframes df1 e df2 para arrays. Multiplique df1 pelo transposto de df2 (use a função que calcula transposta de matriz) e printe o resultado final. Lembre-se de dropar a coluna target_2 de cada um dos dataframes antes de fazer a mutliplicação.

__Exercício 5:__ 
    
5.1: Crie uma função que receba um  número inteiro positivo N e retorne um array 0, 1, ..., N. Dica: Use as funções que vimos para criação de arrays;
    
5.2: Crie uma função que receba um array [0,1, .., N] e retorne o mesmo array, só que com os números em posições aleatórias. Dica: use a função que vimos em aula que faz escolhas aleatórias SEM reposição;

5.3: Crie uma função que receba um número inteiro positivo N. Essa função deve retornar um array de tamanho N, cujos valores são números inteiros entre 0 e N-1 e pode haver repetição desses números.

__Exercício 6:__ Construa uma função divide_dados:
        
        - recebe um array X de dimensões quaisquer (n,k) e um parâmetro p entre 0 e 1.
        - divide o array X em duas partes disjuntas de forma aleatória. Ou seja, crie dois novos arrays, X1 e X2,
        onde a dimensão de linhas de X1 somada com a dimensão de linhas de X2 deve ser igual a dimensão de linhas de X.
        - interpretando p como percentual, o tamanho de X1 deve ser p*100% do número de linhas de X; enquanto X2 deve ser o resto.
        - retorne esses dois arrays.
        
       